frame/staking/src/lib.rs

```diff
@@ -732,6 +731,143 @@ impl<AccountId, Balance: Default + HasCompact> Default for Exposure<AccountId, B
 	}
 }
 
+impl<AccountId: Clone, Balance: HasCompact + AtLeast32BitUnsigned + Copy>
+	Exposure<AccountId, Balance>
+{
+	/// Splits an `Exposure` into `ExposureOverview` and multiple chunks of `IndividualExposure`
+	/// with each chunk having maximum of `page_size` elements.
+	fn into_pages(
+		self,
+		page_size: u32,
+	) -> (ExposureOverview<Balance>, Vec<ExposurePage<AccountId, Balance>>) {
+		let individual_chunks = self.others.chunks(page_size as usize);
+		let mut exposure_pages: Vec<ExposurePage<AccountId, Balance>> =
+			Vec::with_capacity(individual_chunks.len());
+
+		for chunk in individual_chunks {
+			let mut page_total: Balance = Zero::zero();
+			for individual in chunk.iter() {
+				page_total = page_total.saturating_add(individual.value);
+			}
+
+			exposure_pages.push(ExposurePage {
+				page_total,
+				others: chunk
+					.iter()
+					.map(|c| IndividualExposure { who: c.who.clone(), value: c.value })
+					.collect(),
+			});
+		}
+
+		(
+			ExposureOverview {
+				total: self.total,
+				own: self.own,
+				nominator_count: self.others.len() as u32,
+				page_count: exposure_pages.len() as PageIndex,
+			},
+			exposure_pages,
+		)
+	}
+}
+
+/// A snapshot of the stake backing a single validator in the system.
+#[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Encode, Decode, RuntimeDebug, TypeInfo)]
+pub struct ExposurePage<AccountId, Balance: HasCompact> {
+	/// The total balance of this chunk/page.
+	#[codec(compact)]
+	pub page_total: Balance,
+	/// The portions of nominators stashes that are exposed.
+	/// TODO(ank4n): BoundedVec touches lot of code, skip for now.
+	pub others: Vec<IndividualExposure<AccountId, Balance>>,
+}
+
+impl<AccountId, Balance: Default + HasCompact> Default for ExposurePage<AccountId, Balance> {
+	fn default() -> Self {
+		Self { page_total: Default::default(), others: vec![] }
+	}
+}
+
+/// An overview of stake backing a single validator.
+///
+/// It, in combination with a list of `ExposurePage`s, can be used to reconstruct a full `Exposure`
+/// struct. This is useful for cases where we want to query a single page of `Exposure`s.
+#[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Encode, Decode, RuntimeDebug, TypeInfo)]
+pub struct ExposureOverview<Balance: HasCompact> {
+	/// The total balance backing this validator.
+	#[codec(compact)]
+	pub total: Balance,
+	/// The validator's own stash that is exposed.
+	#[codec(compact)]
+	pub own: Balance,
+	/// Number of nominators backing this validator.
+	pub nominator_count: u32,
+	/// Number of pages of backers.
+	pub page_count: PageIndex,
```

Comment:
Are you familiar with `decode_len`? It might be useful in some places, and *might* be enough reason to not store this (just a thought).

Reply:
Pages in `ErasStakersPaged` is stored as a key-value pair and not a vector. But I am looking into NMap again and I think it is possible to pass two keys and iterate over the rest of the keys to get number of pages. That would still mean few extra key reads. 

I still think I should make the EraStakersPaged as an NMap and have the page count check in `try_state` check. But if we keep page_count here, it will save some reads. Wdyt?

I will clean out the `nominator_count`. Few things I added thinking down the rabbit hole it might be useful but this wasn't.